### ISSUE TITLE
foundry optimzer setting

### ISSUE THREAD

**Issue by alxiong** [EspressoSystems/espresso-sequencer#1982]
Currently, contract compilation is too slow and hurts dev experience. 

We should remove these from default profile to speed things up

https://github.com/EspressoSystems/espresso-sequencer/blob/50e0368f3efd7485cbfecf687c066a46a4b3f1fc/foundry.toml#L12-L14

I believe the reason we set `via-ir = true` to work around "stack too deep" problem, but there are many other ways to reduce local variables and stack usage.

additionally, we should probably have a profile for deployments that has higher optimizer run?

**Comment by sveitser** [comment]
Yes the via-ir was turned on for avoiding a "stack too deep" error.

**Comment by alxiong** [comment]
do we want to work on it before mainnet? @philippecamacho @alysiahuggins

**Comment by philippecamacho** [comment]
Should not we reaudit some of the contracts if we do that?

**Comment by alxiong** [comment]
Let me clarify. I simply suggest that maybe we should set "optimizer=x" for production profile for better gas. Not necessarily changing any code, so won't be affected by audit timeline
